tests/CMakeLists.txt

```diff
@@ -20,6 +20,11 @@ if(NOT MSVC)
   set(UNITTEST_STATIC_LINK ON)
 endif()
 
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
This should be part of the BEGIN_WHOLE_ARCHIVE/E#ND_WHOLE_ARCHIVE definition in the top-level CMakeLists.txt

Reply:
Sure, done.